ee/server/service/certificate_authorities.go

```diff
@@ -372,3 +380,209 @@ func (svc *Service) DeleteCertificateAuthority(ctx context.Context, certificateA
+		if err := validateCAName(*hydrant.Name, errPrefix); err != nil {
+			return err
+		}
+	}
+	if hydrant.URL != nil {
+		if err := validateURL(*hydrant.URL, "Hydrant", errPrefix); err != nil {
+			return err
+		}
+	}
+	if hydrant.ClientID != nil && *hydrant.ClientID == "" {
+		return &fleet.BadRequestError{
+			Message: fmt.Sprintf("%sInvalid Hydrant Client ID. Please correct and try again.", errPrefix),
+		}
+	}
+	if hydrant.ClientSecret != nil && *hydrant.ClientSecret == "" {
+		return &fleet.BadRequestError{
+			Message: fmt.Sprintf("%sInvalid Hydrant Client Secret. Please correct and try again.", errPrefix),
+		}
+	}
+	return nil
+}
+
+func (svc *Service) validateNDESSCEPProxyUpdate(ctx context.Context, ndesSCEP *fleet.NDESSCEPProxyCAUpdatePayload, errPrefix string) error {
+	// some methods in this fuction require the NDESSCEPProxyCA type so we convert the ndes update payload here
+	NDESProxy := fleet.NDESSCEPProxyCA{
+		URL:      *ndesSCEP.URL,
+		AdminURL: *ndesSCEP.AdminURL,
+		Username: *ndesSCEP.Username,
+		Password: *ndesSCEP.Password,
+	}
+
+	if ndesSCEP.URL != nil {
+		if err := validateURL(*ndesSCEP.URL, "NDES SCEP", errPrefix); err != nil {
+			return err
+		}
+	}
+	if err := svc.scepConfigService.ValidateSCEPURL(ctx, *ndesSCEP.URL); err != nil {
+		level.Error(svc.logger).Log("msg", "Failed to validate NDES SCEP URL", "err", err)
+		return &fleet.BadRequestError{Message: fmt.Sprintf("%sInvalid SCEP URL. Please correct and try again.", errPrefix)}
+	}
+	if err := svc.scepConfigService.ValidateNDESSCEPAdminURL(ctx, NDESProxy); err != nil {
+		level.Error(svc.logger).Log("msg", "Failed to validate NDES SCEP admin URL", "err", err)
+		switch {
+		case errors.As(err, &NDESPasswordCacheFullError{}):
+			return &fleet.BadRequestError{Message: fmt.Sprintf("%sThe NDES password cache is full. Please increase the number of cached passwords in NDES and try again.", errPrefix)}
+		case errors.As(err, &NDESInsufficientPermissionsError{}):
+			return &fleet.BadRequestError{Message: fmt.Sprintf("%sInsufficient permissions for NDES SCEP admin URL. Please correct and try again.", errPrefix)}
+		default:
+			return &fleet.BadRequestError{Message: fmt.Sprintf("%sInvalid NDES SCEP admin URL or credentials. Please correct and try again.", errPrefix)}
+		}
+	}
+	return nil
+}
+
+func (svc *Service) validateCustomSCEPProxyUpdate(ctx context.Context, customSCEP *fleet.CustomSCEPProxyCAUpdatePayload, errPrefix string) error {
+	if customSCEP.Name != nil {
+		if err := validateCAName(*customSCEP.Name, errPrefix); err != nil {
+			return err
+		}
+	}
+	if customSCEP.URL != nil {
+		if err := validateURL(*customSCEP.URL, "SCEP", errPrefix); err != nil {
+			return err
+		}
+	}
+	if customSCEP.Challenge != nil && *customSCEP.Challenge == "" {
+		return &fleet.BadRequestError{
+			Message: fmt.Sprintf("%sCustom SCEP Proxy challenge cannot be empty", errPrefix),
+		}
+	}
+	if err := svc.scepConfigService.ValidateSCEPURL(ctx, *customSCEP.URL); err != nil {
```

Comment:
If we do not alter the URL it will be empty in the update payload, and this will panic/fail. We need to guard it with a URL != nil check.